web/react/components/audio_video_preview.jsx

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import Constants from '../utils/constants.jsx';
+import FileInfoPreview from './file_info_preview.jsx';
+import * as Utils from '../utils/utils.jsx';
+
+export default class AudioVideoPreview extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.handleFileInfoChanged = this.handleFileInfoChanged.bind(this);
+        this.handleLoadError = this.handleLoadError.bind(this);
+
+        this.stop = this.stop.bind(this);
+
+        this.state = {
+            canPlay: true
+        };
+    }
+
+    componentWillMount() {
+        this.handleFileInfoChanged(this.props.fileInfo);
+    }
+
+    componentDidMount() {
+        if (this.refs.source) {
+            $(ReactDOM.findDOMNode(this.refs.source)).one('error', this.handleLoadError);
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.fileUrl !== nextProps.fileUrl) {
+            this.handleFileInfoChanged(nextProps.fileInfo);
+        }
+    }
+
+    handleFileInfoChanged(fileInfo) {
+        let video = ReactDOM.findDOMNode(this.refs.video);
+        if (!video) {
+            video = document.createElement('video');
+        }
+
+        const canPlayType = video.canPlayType(fileInfo.mime_type);
+
+        this.setState({
+            canPlay: canPlayType === 'probably' || canPlayType === 'maybe'
```

Comment:
I don't think using constants really adds anything to the code there since "probably" and "maybe" are part of the HTML5 spec for that function. I'd just end up with constants called something like CAN_PLAY_PROBABLY and CAN_PLAY_MAYBE. Thoughts?